assignment_2.rb

```diff
@@ -0,0 +1,68 @@
+class TreeNode 
+  attr_accessor :val,:l,:r 
+
+    def initialize(val, l=nil , r=nil)
+      @val = val
+      @l = l 
+      @r = r
+    end 
+end
+  
+# max = root node 
+def traverse(tree, max)
+  sum = 0
+  #check if val is higher than root
+  if tree.val >= max 
+    sum = 1 # if tree is valid, always count root 
+    sum += traverse(tree.l,max) if tree.l != nil 
```

Comment:
Maybe we should pass `tree.val` instead of `max` in this case?
```
tree_l_l = TreeNode.new(2)
tree_l = TreeNode.new(3, tree_l_l)
tree = TreeNode.new(1, tree_l)
p solution(tree)
```
This should return 2. Since `tree_l_l` is not visible because value of `tree_l` is 3